core/trino-main/src/main/java/io/trino/sql/planner/plan/TableFunctionNode.java

```diff
@@ -114,29 +144,47 @@ public <R, C> R accept(PlanVisitor<R, C> visitor, C context)
     public PlanNode replaceChildren(List<PlanNode> newSources)
     {
         checkArgument(sources.size() == newSources.size(), "wrong number of new children");
-        return new TableFunctionNode(getId(), name, arguments, properOutputs, newSources, tableArgumentProperties, handle);
+        return new TableFunctionNode(getId(), name, arguments, properOutputs, newSources, tableArgumentProperties, copartitioningLists, handle);
     }
 
     public static class TableArgumentProperties
     {
+        private final String argumentName;
+        private final Multimap<String, Symbol> columnMapping;
         private final boolean rowSemantics;
         private final boolean pruneWhenEmpty;
         private final boolean passThroughColumns;
-        private final Specification specification;
+        private final Optional<Specification> specification;
```

Comment:
It's strange for a table function to use a `WindowNode.Specification`. We should either move it out of WindowNode (preferable) or create a separate one (not ideal if they represent the same concept)

Reply:
Extracted `Specification`.